typespec-extension/src/code-model-builder.ts

```diff
@@ -1341,7 +1346,7 @@ export class CodeModelBuilder {
       schema = this.processSchemaFromSdkType(sdkType, body.name);
     }
 
-    const isAnonymousModel = sdkType.kind === "model" && sdkType.isGeneratedName === true;
+    const isAnonymousModel = sdkType.kind === "model" && body.kind === "Model" && !this.isArm();
```

Comment:
why do we remove the condition `sdkType.isGeneratedName === true` for checking anonymous model?
"

Reply:
Because this is no longer about anonymous model.

The feature ask is to treat `op(...Model)` the same, regardless of `alias Model` or `model Model`

Reply:
Rename the var to `bodyParameterFlatten`. Also added comment to make the logic understandable.